sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -1036,22 +1037,45 @@ def predict_log_proba(self, X):
         y_prob = self.predict_proba(X)
         return np.log(y_prob, out=y_prob)
 
-    def predict_proba(self, X):
+    def predict_proba(self, X, max_row_chunk=None):
         """Probability estimates.
 
         Parameters
         ----------
         X : {array-like, sparse matrix}, shape (n_samples, n_features)
             The input data.
 
+        max_row_chunk: integer
+            The number of rows to use at a single predict call. If None optimal 
+            row number fit that fits the working_memory is used.
+
         Returns
         -------
         y_prob : array-like, shape (n_samples, n_classes)
             The predicted probability of the sample for each class in the
             model, where classes are ordered as they are in `self.classes_`.
         """
         check_is_fitted(self, "coefs_")
-        y_pred = self._predict(X)
+        # split and collate depending on the working_memory
+        x_axis0_size = X.shape[0]
+        if not max_row_chunk:
+            chunk_size = get_chunk_n_rows(row_bytes=sys.getsizeof(X[0:1]),
+                                          max_n_rows=x_axis0_size)
+        else:
+            chunk_size = max_row_chunk
+        # Pre-allocate the entire result set to avoid array copying for
+        # efficiency. As we do not know the shape of the output,
+        # we simply test the output size for a single sample
+        # and use the shape of the output
+        y_pred = np.empty((x_axis0_size,) + self._predict(X[0:1]).shape[1:])
```

Comment:
This is problematic when we want to predict a single sample (this would run the predict twice). Can't we determine the size of the output without running the predict?

Reply:
Actually, we are having an overhead of +1 calls, no doubt it is 100% when its only 1 sample. But I think it is quite acceptable bottle-neck for the first iteration.
Please let me know your views